internal/httpcli/client.go

```diff
@@ -366,9 +367,10 @@ func NewLoggingMiddleware(logger log.Logger) Middleware {
 	}
 }
 
-//
 // Common Opts
-//
+var hostList = env.Get("EXTERNAL_DENY_LIST", "", "Deny list for outgoing requests")
+var denyList = hostmatcher.ParseHostMatchList("", hostList)
+var once sync.Once
```

Comment:
Makes it clearer what it's protecting and usually what we do:

```suggestion
var denyListOnce sync.Once
```